buttercup/cogs/find.py

```diff
@@ -1,69 +1,238 @@
+from discord.ext.commands import Cog
+from discord_slash import SlashContext, cog_ext
+from discord_slash.utils.manage_commands import create_option
+
+from buttercup.bot import ButtercupBot
+
+PLEASE_CHECK_URL = (
+    " Please check that your link is correct, it should lead to"
+    " either a post on r/TranscribersOfReddit, a post on a partner"
+    " sub or to a transcription."
+)
+DISCORD_USERNAME_LINK = "[{0}](https://reddit.com/u/{0})"
+
+
+def normalize_url(reddit_url_str: str) -> Union[str, None]:
+    """
+    Normalize a Reddit URL to the format that Blossom uses.
+
+    This is necessary because the link could be to Old Reddit, for example.
+    """
+    parse_result = urlparse(reddit_url_str)
+    if "reddit" not in parse_result.netloc:
+        return None
+
+    # On Blossom, all URLs end with a slash
+    path = parse_result.path
+
+    if not path.endswith("/"):
+        path += "/"
+
+    return f"https://reddit.com{path}"
+
+
+def get_id_from_url(grafeas_url: str) -> int:
+    """Extract the API from a Grafeas URL."""
+    return int(grafeas_url.split("/")[-2])
+
+
+def limit_str(text: str, limit: Optional[int] = None) -> str:
+    """Limit the string to the given length."""
+    if limit is None or len(text) <= limit:
+        return text
+
+    return f"{text[:(limit - 3)]}..."
+
+
+class Find(Cog):
+    def __init__(self, bot: ButtercupBot, blossom_api: BlossomAPI) -> None:
+        """Initialize the Find cog."""
+        self.bot = bot
+        self.blossom_api = blossom_api
+
+    def to_embed(self) -> Embed:  # noqa: C901
+        """Convert the submission to a Discord embed."""
+        color = Color.from_rgb(255, 176, 0)  # Orange
+        status: str = "Unclaimed"
+
+        tr_link = f"[Link]({self.transcription['url']})" if self.transcription else None
+
+        if self.submission["completed_by"]:
+            color = Color.from_rgb(148, 224, 68)  # Green
+
+            if self.author:
+                status = "Completed by {}".format(
+                    DISCORD_USERNAME_LINK.format(self.author["username"])
+                )
+            else:
+                status = "Completed"
+        elif self.submission["claimed_by"]:
+            color = Color.from_rgb(13, 211, 187)  # Cyan
+
+            if self.author:
+                status = "Claimed by {}".format(
+                    DISCORD_USERNAME_LINK.format(self.author["username"])
+                )
+            else:
+                status = "Claimed"
+
+        embed = (
+            Embed(color=color)
+            .add_field(name="Status", value=status)
+            .add_field(
+                name="OCR",
+                value="Yes" if self.submission["has_ocr_transcription"] else "No",
+            )
+            .add_field(
+                name="Archived", value="Yes" if self.submission["archived"] else "No"
+            )
+        )
+
+        if self.transcription:
+            embed.description = limit_str(self.transcription["text"], 200)
```

Comment:
Before, the header and footer would be cut off here to make it smaller. Here the special case of OCR has to be considered where no header and footer is present.

Reply:
Ported your code over, so this one is fixed.